Source/Core/Core/Debugger/BranchWatch.cpp

```diff
@@ -0,0 +1,312 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/Debugger/BranchWatch.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdio>
+
+#include <fmt/format.h>
+
+#include "Common/Assert.h"
+#include "Common/BitField.h"
+#include "Common/CommonTypes.h"
+#include "Core/Core.h"
+#include "Core/PowerPC/Gekko.h"
+#include "Core/PowerPC/MMU.h"
+
+namespace Core
+{
+void BranchWatch::Clear(const CPUThreadGuard&)
+{
+  m_selection.clear();
+  m_collection_vt.clear();
+  m_collection_vf.clear();
+  m_collection_pt.clear();
+  m_collection_pf.clear();
+  m_recording_phase = Phase::Blacklist;
+  m_blacklist_size = 0;
+}
+
+union USnapshotMetadata
```

Comment:
This could also use a comment for what exactly it's designed for.

Reply:
I have now added one.

Reply:
Yeah that's a lot better, thanks.